workflow/sync/sync_manager.go

```diff
@@ -101,8 +178,13 @@ func (cm *Manager) Initialize(wfs []wfv1.Workflow) {
 				if mutex == nil {
 					mutex := cm.initializeMutex(holding.Mutex)
 					if holding.Holder != "" {
-						resourceKey := getResourceKey(wf.Namespace, wf.Name, holding.Holder)
-						mutex.acquire(resourceKey)
+						level, err := getWorkflowSyncLevelByName(&wf, holding.Mutex)
+						if err != nil {
+							log.Warnf("cannot obtain lock level for %s", holding.Mutex)
+							continue
+						}
```

Comment:
Since this could error out, it might be more resilient to actually check if we need to care about the level before calling this function.

Reply:
In the mutex case it cannot be held more than once, so level should always be accurate.

An error indicates a bug here doesn't it?

Reply:
Yeah we shouldn't really hit this error, it indicates a bug in `getWorkflowSyncLevel` or in the functions it calls or in the mutex logic itself.

Reply:
Left it as it is.